xdsl/backend/riscv/riscv_scf_to_asm.py

```diff
@@ -69,15 +69,15 @@ def match_and_rewrite(self, op: riscv_scf.ForOp, rewriter: PatternRewriter, /):
         yield_op = body.last_op
         assert isinstance(yield_op, riscv_scf.YieldOp)
 
-        body.insert_ops_after(
+        rewriter.insert_op(
```

Comment:
What is the benefit of this change?

Reply:
If we don't notify the `PatternRewriter` that these operations are added, they are not matched anymore by the `PatternRewriter`.